napi-inl.h

```diff
@@ -2424,6 +2428,41 @@ inline ClassPropertyDescriptor<T> ObjectWrap<T>::InstanceMethod(
 }
 
 template <typename T>
+inline ClassPropertyDescriptor<T> ObjectWrap<T>::InstanceMethod(
+    Symbol name,
+    InstanceVoidMethodCallback method,
+    napi_property_attributes attributes,
+    void* data) {
+  // TODO: Delete when the class is destroyed
+  InstanceVoidMethodCallbackData* callbackData =
+    new InstanceVoidMethodCallbackData({ method, data});
+
+  napi_property_descriptor desc = {};
+  desc.name = name;
+  desc.method = T::InstanceVoidMethodCallbackWrapper;
+  desc.data = callbackData;
+  desc.attributes = attributes;
+  return desc;
+}
+
+template <typename T>
+inline ClassPropertyDescriptor<T> ObjectWrap<T>::InstanceMethod(
+    Symbol name,
+    InstanceMethodCallback method,
+    napi_property_attributes attributes,
+    void* data) {
+  // TODO: Delete when the class is destroyed
```

Comment:
Wondering about the TODO's here and in the other place.  Is there some reason we can't add the deletion as part of this PR ?

Reply:
Defining a class just creates a constructor function, I'm not sure how can we attach a cleanup action when it is garbage collected?

An alternative would be to use `data` for a pointer to the method callback and disallow an additional `void* data` in the C++ API; although that still wouldn't solve the case for the getter/setter (two pointers instead of one).

Reply:
I see this is not specific to your change, as there were already other instances.